modules/spring-config/src/main/java/org/mule/runtime/config/internal/context/SpringMuleContextServiceConfigurator.java

```diff
@@ -467,12 +480,27 @@ protected MemoryManagementService getMemoryManagementService() {
     return memoryManagementService;
   }
 
-  private static BeanDefinition resolveMeterProvider() {
+  private BeanDefinition resolveArtifactMeterProvider() {
```

Comment:
If this is for decorating with the artifactId and supposedly in the context of an artifact you don't want to decorate with the artifactId, I would change the name indicating this is for adding the artifactId as attribute.